File: Part 2 - Regression/Random Forest Regression/Random_Forest.R
#importing data to r inv
data=read.csv("Position_Salaries.csv")
dataset=data[2:3]

#importing lib
install.packages("randomForest")
library("randomForest")
req=randomForest(x=dataset[1],
                 y=dataset$Salary,
                 ntree = 500)
#prediction
y_pred=predict(req,data.frame(Level=6.5))

#visualising
library("ggplot2")
X_grid=seq(min(dataset$Level),max(dataset$Level),0.1)
ggplot()+
  geom_point(aes(x=dataset$Level,y=dataset$Salary),
             color='red')+
  geom_line(aes(x=X_grid,y=predict(req,newdata = data.frame(Level=X_grid))),
            color="blue")+
  ggtitle("RandomForestReg")+
  xlab("Position")+
  ylab("Salary")

File: Part 2 - Regression/Random Forest Regression/Random_Forest.py
#importing packages
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

#importing data

data=pd.read_csv("Position_Salaries.csv")
X=data.iloc[:,1:2].values
y=data.iloc[:, 2].values

#building the model
from sklearn.ensemble import RandomForestRegressor
req=RandomForestRegressor(n_estimators=10,random_state=0)
req.fit(X,y)

#prediction
y_prid=req.predict(6.5)

#visu
X_grid=np.arange(min(X),max(X),0.01)
X_grid=X_grid.reshape((len(X_grid),1))
plt.scatter(X,y,color="red")
plt.plot(X_grid,req.predict(X_grid),color='green')
plt.title("D.T.Regressor")
plt.xlabel("Position")
plt.ylabel("Salary")
plt.show()
